cdc/capture/capture.go

```diff
@@ -283,10 +283,28 @@ func (c *Capture) campaignOwner(ctx cdcContext.Context) error {
 			return cerror.ErrCaptureSuicide.GenWithStackByArgs()
 		}
 
-		log.Info("campaign owner successfully", zap.String("capture-id", c.info.ID))
+		ownerRev, err := c.etcdClient.GetOwnerRevision(ctx, c.info.ID)
+		if err != nil {
+			if errors.Cause(err) == context.Canceled {
+				return nil
+			}
+			return errors.Trace(err)
+		}
+
+		// We do a copy of the globalVars here to avoid
+		// accidental modifications and potential race conditions.
+		globalVars := *ctx.GlobalVars()
```

Comment:
The struct `GlobalVars` share some pointer fields. Do we need to implement the deep copy for the `GlobalVars`?

Reply:
I don't think so. Those are globally shared resources such as the pd client, the tikv client, etc, and they do not need to be copied.